lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -1056,15 +1065,17 @@ int construct_pdu_session_reject_dl_req(
   if (is_security_enabled) {
     msg->security_protected.plain.amf.header.extended_protocol_discriminator =
         M5G_MOBILITY_MANAGEMENT_MESSAGES;
-    msg->security_protected.plain.amf.header.message_type = DLNASTRANSPORT;
+    msg->security_protected.plain.amf.header.message_type =
+        static_cast<M5GMessageType>(M5GMessageType::DLNASTRANSPORT);
     msg->header.security_header_type =
         SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_CYPHERED;
     dlmsg = &msg->security_protected.plain.amf.msg.downlinknas5gtransport;
   } else {
     msg->plain.amf.header.extended_protocol_discriminator =
         M5G_MOBILITY_MANAGEMENT_MESSAGES;
-    msg->plain.amf.header.message_type = DLNASTRANSPORT;
-    msg->header.security_header_type   = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+    msg->plain.amf.header.message_type =
+        static_cast<M5GMessageType>(M5GMessageType::DLNASTRANSPORT);
```

Comment:
What's the need to cast from "M5GMessageType" as "M5GMessageType"?

Reply:
addressed review comment